🧪 Pre-build armv7l images for building wheels (clean attempt)

webknjaz opened the pull request:
These are relatively recent upstream and only exist for glibc 2.31+.

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
$sbj.

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Maintenance Pull Request
- Packaging Pull Request

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
Blocks #648.

alex committed:
build armv7l images (#669)

* build armv7l images

* Update build-docker-images.yml

* Don't use `enable-ec_nistp_64_gcc_128` on armv7l

* hack?

* We are an armv4 build

(cherry picked from commit 519b16bd0803ea22dbd0a4525c641060581c0215)

webknjaz committed:
🧪 Pre-build armv7l images for building wheels

These are relatively recent upstream and only exist for glibc 2.31.

Jakuje commented:
This one fails already to build openssl:

> crypto/ec/ecp_nistp224.c:49:3: error: #error "Your compiler doesn't appear to support 128-bit integer types"

You need the same workaround as for s390x (exclude the `enable-ec_nistp_64_gcc_128`)

webknjaz commented:
@Jakuje I've cherry-picked https://github.com/pyca/infra/commit/519b16bd0803ea22dbd0a4525c641060581c0215#diff-7bd2e32999fd3511ce4aa7073dad04d03ac90be5c97b9197d40beb0042df7802.

Actually, I did that already in #719 but didn't drag it into this PR originally. Now that it's in, the libdl errors are visible yet again.

webknjaz committed:
Build libssh without examples

Apparently, building the examples somehow attempts linking against
`libdl` that it can't locate. But we don't need them regardless.

webknjaz committed:
📝 Add a change note for PR #740
